src/bgen/BindingTouch.cs

```diff
@@ -334,13 +317,49 @@ int Main3 (string [] args)
 				}
 			}
 
-			var api = TypeManager.ParseApi (apiAssembly, config.ProcessEnums);
+			api = TypeManager.ParseApi (apiAssembly, config.ProcessEnums);
 			namespaceCache ??= new NamespaceCache (
 				CurrentPlatform,
 				config.HelperClassNamespace ?? firstApiDefinitionName,
 				LibraryManager.DetermineSkipSystemDrawing (LibraryInfo.TargetFramework)
 			);
 
+
+		} catch (Exception ex) {
+			ErrorHelper.Show (ex);
+		}
+
+		return true;
+	}
+
+	int Main3 (string [] args)
+	{
+		ErrorHelper.ClearWarningLevels ();
+		BindingTouchConfig config = new ();
+
+		if (!TryCreateOptionSet (config, args))
+			return 1;
+
+		if (config.ShowHelp) {
+			ShowHelp (config.OptionSet);
+			return 0;
+		}
+
+		libraryInfo = LibraryInfo.LibraryInfoBuilder.Build (references, config);
+		CurrentPlatform = LibraryManager.DetermineCurrentPlatform (TargetFramework.Platform);
+
+		if (!TryInitializeApi (config, out Api? api) || !TryGenerate (config, api))
+			return 1;
+
+		return 0;
+	}
+
+	bool TryGenerate (BindingTouchConfig config, Api? api)
+	{
+		if (api is null)
+			return false;
+
```

Comment:
With the [NotNullWhen (true)] attribute in TryInitializeApi, the Api parameter doesn't need to be nullable anymore:

```suggestion
	bool TryGenerate (BindingTouchConfig config, Api api)
	{
```